common/marshal/marshal.go

```diff
@@ -34,7 +34,11 @@ func MarshalJsonTx(tx *types.Tx, params *params.Params, blkHashStr string,
 	if tx == nil {
 		return json.TxRawResult{}, errors.New("can't marshal nil transaction")
 	}
-	return MarshalJsonTransaction(tx.Transaction(), params, blkHashStr, confirmations)
+	txr, err := MarshalJsonTransaction(tx.Transaction(), params, blkHashStr, confirmations)
+	if err == nil {
+		txr.Duplicate = tx.IsDuplicate
```

Comment:
why the additional logic here?  why the modification have to be outside of the ‘MashalJsonTranscation’ method? or after the method calling? should it happen before or inside of the method invoke?

Reply:
> why the additional logic here? why the modification have to be outside of the ‘MashalJsonTranscation’ method? or after the method calling? should it happen before or inside of the method invoke?

I think the direct reason is that IsDuplicate is only inside types.Tx rather than types.Transaction, types.Tx  is just a wrapper of raw transaction ( types.Transaction)